### DISCUSSION TITLE
Linux 以外の OS への対応

### DISCUSSION THREAD

**Discussion by tokuhirom** [akaza-im/akaza#186] [Ideas]
Linux のみ、となるとやはり母数が少なく、利用者が限定的になってしまうので、Mac や Windows などに対応したほうがよいのではないだろうか。

**Comment by omasanori** [comment]
Windows版開発に役立ちそうなリンク：

* [Input Method Editor (IME) requirements - Windows apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/apps/design/input/input-method-editor-requirements "Input Method Editor (IME) requirements - Windows apps | Microsoft Learn")
* [Text Services Framework (Text Services Framework) - Win32 apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/win32/tsf/text-services-framework "Text Services Framework (Text Services Framework) - Win32 apps | Microsoft Learn")
* [microsoft/windows-rs: Rust for Windows](https://github.com/microsoft/windows-rs "microsoft/windows-rs: Rust for Windows")
* [WinAnthy](https://www.kmc.gr.jp/projects/winanthy/ "WinAnthy")
* [nathancorvussolis/corvusskk: ▽▼ SKK-like Japanese Input Method Editor for Windows](https://github.com/nathancorvussolis/corvusskk "nathancorvussolis/corvusskk: ▽▼ SKK-like Japanese Input Method Editor for Windows")

**Reply by tokuhirom** [reply]
ありがとうございます。macよりwindowsのほうが情報多そうですね

**Comment by tokuhirom** [comment]
Mac の IM だと [gyaim](https://masui.github.io/GyaimMotion/) ぐらいしか情報がなさそう。
あとは中国語の IM は OSS でも対応しているものが多数あるので、そのへんから情報を得るのがよいかなぁ。。 [rime](https://rime.im/) とか

**Reply by omasanori** [reply]
日本語入力の例は[AquaSKK](https://github.com/codefirst/aquaskk)があります。APIは[Input Method Kit](https://developer.apple.com/documentation/inputmethodkit)です。

**Reply by tokuhirom** [reply]
あー。AquaSKK! 僕も昔使っていましたがすっかり存在を忘れていました

**Reply by tokuhirom** [reply]
個人的には$dayjobではmacを使ってるので、mac対応出来ると仕事でも使えていいなーと思っています。
linuxはデスクトップを触っている時間は、akazaの開発をメインでしてる時間が長くて日本語を入力している時間が短いので。

**Reply by tokuhirom** [reply]
[日本語入力を作るのに必要だった本](https://amzn.to/3H9E7J0) も参考になりそうです。

**Reply by tokuhirom** [reply]
https://github.com/mzp/EmojiIM

**Reply by tokuhirom** [reply]
https://github.com/toshi-pono/TypoIME https://speakerdeck.com/toshi00/traplt0327
rust で書かれていてよさそう。

**Reply by tokuhirom** [reply]
TypoIME をベースにプロトタイピングしてみたが、rust で全てを書くのはなかなかむつかしいということがわかりました。
